### PR TITLE
BluetoothManager refactor

### PR DESCRIPTION
**Pull request by anna1901**
# What was done?
This PR includes refactor of two big classes that were responsible for managing mobile bluetooth session: `BluetoothManager` and `MobilePeripheralSessionManager`.
`BluetoothManager` was replaced with a `NewBluetoothManager` class (probably should be renamed soon), which now is the only class in the project which is using CoreBluetooth.
Logic from `MobilePeripheralSessionManager` was split into separate services and controllers:
* `MeasurementsRecordingServices` - responsible for receiving measurements from AirBeam, parsing them into a stream and passing them further in completion block.
* `ActiveMobileSessionProvidingService` for storing active mobile session
* `BluetoothSessionRecordingController` responsible for starting, resuming and stopping recording
* `MeasurementsSavingService` for creating session and saving measurements to the db

## Reconnecting and Standalone Mode
There is also a new reconnection mechanism which is managed by `DefaultReconnectionController` which is informed by `NewBluetoothManager` that the device has disconnected, it is performing reconnection attempt and then is informing its delegate, `SessionManagingReconnectionController`, about the result.

All the changes related with moving session to standalone mode are done by `DefaultStandaloneModeContoller`

## Additional changes:
* `AirBeam3Configurator` was refactored and now it calls a completion to let the calling class know whether configuration call to AB was successful.
* check for device compatibility was added to `ConnectingAirBeamServices` (we are checking discovered characteristics)
* `SelectPeripheralViewModel` was added
* If AB disconnects from the phone during SD sync it gets detected and an appropriate alert is shown

### CODE REVIEW

File: AirCastingTests/Logic/CreatingSession/SelectPeripheralViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    func testExample() throws {
+        // This is an example of a functional test case.
+        // Use XCTAssert and related functions to verify your tests produce the correct results.
+        // Any test you write for XCTest can be annotated as throws and async.
+        // Mark your test throws to produce an unexpected failure when your test encounters an uncaught error.
+        // Mark your test async to allow awaiting for asynchronous code to complete. Check the results with assertions afterwards.
+    }
+}
```
Comment (jankrzempek) [inline AirCastingTests/Logic/CreatingSession/SelectPeripheralViewModel.swift:23]: Should still exists?

File: AirCastingTests/Logic/BluetoothSession/DefaultStandaloneModeContollerTests.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        case start
+        case stop
+    }
+    var callHistory: [HistoryItem] = []
+    var location: CurrentValueSubject<CLLocation?, Never> = .init(nil)
+    func start() {
+        callHistory.append(.start)
+    }
+    func stop() {
+        callHistory.append(.start)
```
Comment (jankrzempek) [inline AirCastingTests/Logic/BluetoothSession/DefaultStandaloneModeContollerTests.swift:73]: I think here should be `.stop`

File: AirCastingTests/Logic/BluetoothSession/MobileAirBeamSessionRecordingControllerTests.swift [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+final class MobileAirBeamSessionRecordingControllerTests: ACTestCase {
+    let sut = MobileAirBeamSessionRecordingController()
+    private var measurementsSaver = MeasurementsSavingServiceMock()
+    private var storage = MobileSessionStorageMock()
+    private var measurementsRecorder =  MeasurementsRecordingServicesMock()
+    private var activeSessionProvider = ActiveMobileSessionProvidingServiceMock()
+    private var locationTracker = LocationTrackerMock()
+    private var btManager = BluetoothConnectionHandlerMock()
+//    private var configurator = AirBeamConfiguratorMock(device: <#NewBluetoothManager.BluetoothDevice#>)
```
Comment (jankrzempek) [inline AirCastingTests/Logic/BluetoothSession/MobileAirBeamSessionRecordingControllerTests.swift:17]: commented, can be deleted?

File: AirCastingTests/Logic/BluetoothSession/MobileAirBeamSessionRecordingControllerTests.swift [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    
+    override func setUp() {
+        super.setUp()
+        Resolver.test.register { self.measurementsSaver as MeasurementsSavingService }
+        Resolver.test.register { self.storage as MobileSessionStorage }
+        Resolver.test.register { self.measurementsRecorder as MeasurementsRecordingServices }
+        Resolver.test.register { self.activeSessionProvider as ActiveMobileSessionProvidingService }
+        Resolver.test.register { self.locationTracker as LocationTracker }
+        Resolver.test.register { self.btManager as BluetoothConnectionHandler }
+//        Resolver.test.register { self.configurator as AirBeamConfigurator }
```
Comment (jankrzempek) [inline AirCastingTests/Logic/BluetoothSession/MobileAirBeamSessionRecordingControllerTests.swift:27]: commented, can be deleted?

**Review by jankrzempek — COMMENTED** [review]
A lot, really amazing!

File: AirCasting/BluetoothSession/BluetoothSessionRecordingController.swift [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    @Injected private var storage: MobileSessionStorage
+    @Injected private var measurementsRecorder: MeasurementsRecordingServices
+    @Injected private var activeSessionProvider: ActiveMobileSessionProvidingService
+    @Injected private var locationTracker: LocationTracker
+    @Injected private var btManager: BluetoothConnectionHandler
+    
+    func startRecording(session: Session, device: NewBluetoothManager.BluetoothDevice, completion: @escaping (Result<Void, Error>) -> Void) {
+        // Step 1: Configure AB
+        Resolver.resolve(AirBeamConfigurator.self, args: device)
+            .configureMobileSession(location: session.location ?? CLLocationCoordinate2D(latitude: 200, longitude: 200)) { [self] result in
```
[the remark this answers is not in the document]
Comment (anna1901) [reply AirCasting/BluetoothSession/BluetoothSessionRecordingController.swift:25]: Why do you think so?

File: AirCasting/Utils/Bluetooth/BluetoothCommunicator.swift [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import Foundation
+
+struct CharacteristicUUID: Hashable {
+    let value: String
+}
+
+protocol BluetoothCommunicator {
+    typealias CharacteristicObserverAction = (Result<Data?, Error>) -> Void
+    
+    /// Adds an entry to observers of a particular characteristic
```
Comment (pawelgil) [inline AirCasting/Utils/Bluetooth/BluetoothCommunicator.swift:13]: other core protocols could also use this kind of documentation. It not only helps the user, but often helps to clear the vision while developing

**Pull request merged**